core/src/main/java/feign/codec/InterceptorDecoder.java

```diff
@@ -0,0 +1,26 @@
+package feign.codec;
+
+import feign.Response;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+public class InterceptorDecoder implements Decoder {
```

Comment:
ResponseMappingDecoder? Add some javadoc since this is a public type, and tighten up with finals etc?